### PR TITLE
evang analyzer first steps

### PR DESCRIPTION
**Pull request by oisinod**
right now the core code is there to start looking for problems in ramps. next pr will branch off of this one, which will start to display the first set of problems

### CODE REVIEW

File: src/analysis/retail/priest/discipline/modules/guide/EvangelismAnalysis.tsx [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+  checkForWrongCasts(ramp: CastEvent[]) {
+    return ramp
+      .map((cast, index) => {
+        if (!ALLOWED_PRE_EVANG.includes(cast.ability.guid)) {
+          return index;
+        }
+        return null;
+      })
+      .filter((cast) => !isNullOrFalse(cast));
+  }
```
Comment (kfinch) [inline src/analysis/retail/priest/discipline/modules/guide/EvangelismAnalysis.tsx]: A bit confused how this works - looks like `checkForWrongCasts` builds and returns an array of the indexes of wrong casts, but then `analyzeSequence` just discards that array? Nothing is being done with the return value.

File: src/analysis/retail/priest/discipline/modules/guide/EvangelismAnalysis.tsx [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    while (!PEMITTED_RAMP_STARTERS.includes(ramp[0].ability.guid)) {
+      ramp.shift();
+    }
+    this.analyzeSequence(ramp);
+    // this.evangSequences.push({ramp: ramp, details: {nonApplicatorCasts: []}})
+  }
+
+  analyzeSequence(ramp: CastEvent[]) {
+    this.checkForWrongCasts(ramp);
+  }
```
Comment (kfinch) [inline src/analysis/retail/priest/discipline/modules/guide/EvangelismAnalysis.tsx]: Why does this method exist? Its only purpose is to call the next method in the sequence..

File: src/analysis/retail/priest/discipline/modules/guide/EvangelismAnalysis.tsx [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+  checkForWrongCasts(ramp: CastEvent[]) {
+    return ramp
+      .map((cast, index) => {
+        if (!ALLOWED_PRE_EVANG.includes(cast.ability.guid)) {
+          return index;
+        }
+        return null;
+      })
+      .filter(Number);
+  }
```
Comment (kfinch) [inline src/analysis/retail/priest/discipline/modules/guide/EvangelismAnalysis.tsx]: Pretty sure you should be able to make this return type `number[]`, which will make downstream handling less of a pain in the ass

File: src/analysis/retail/priest/discipline/modules/guide/EvangelismAnalysis.tsx [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+  TALENTS_PRIEST.RENEW_TALENT.id,
+  SPELLS.FLASH_HEAL.id,
+  TALENTS_PRIEST.RAPTURE_TALENT.id,
+  SPELLS.POWER_WORD_SHIELD.id,
+];
+
+interface Ramp {
+  timestamp: number;
+  badCastIndexes?: number[];
+}
```
Comment (kfinch) [inline src/analysis/retail/priest/discipline/modules/guide/EvangelismAnalysis.tsx]: You need `rampHistory` here too right? In order for the indices to mean anything

Comment (oisinod) [reply src/analysis/retail/priest/discipline/modules/guide/EvangelismAnalysis.tsx]: Yup, great spot! 

I've added it now :)

**Review by kfinch — APPROVED** [review]

**Pull request merged**